query/client.go

```diff
@@ -0,0 +1,58 @@
+package query
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/babylonchain/rpc-client/config"
+	"github.com/cosmos/cosmos-sdk/client"
+	grpctypes "github.com/cosmos/cosmos-sdk/types/grpc"
+	lensquery "github.com/strangelove-ventures/lens/client/query"
+	rpcclient "github.com/tendermint/tendermint/rpc/client"
+	"google.golang.org/grpc/metadata"
+)
+
+var _ BabylonQueryClient = &QueryClient{}
+
+// QueryClient is a client that can only perform queries to a Babylon node
+// It only requires `Cfg` to have `Timeout` and `RPCAddr`, but not other fields
+// such as keyring, chain ID, etc..
+type QueryClient struct {
+	RPCClient rpcclient.Client
+	Cfg       *config.BabylonConfig
```

Comment:
Here one needs to specify a Babylon config for the query client. However, it seems that we only need the RPC address and the timeout. I suggest we make `Timeout` an attribute of the QueryClient, and on `New()` we provide the RPC address and the timeout as parameters